### ISSUE TITLE
Suggestion - self.file alongside self.template that serves a single static file to a given URL.

### ISSUE THREAD

**Issue by drforr** [Bailador/Bailador#144]
It's really a synonym for self.template, it just explicitly doesn't do any preprocessing to the text. No real speed improvements, just clarity, being able to say that route X is completely static, no preprocessing.

--- cut here ---
	submethod BUILD(|) {
		my $rootdir = $?FILE.IO.parent.parent;
		self.get: '/' => sub {
			self.file: 'index.html'; # <--- There.
		};
	}
--- cut here ---

**Comment by szabgab** [comment]
IMHO this should automatically set the content-type as well.
Question: should this be restricted to the directory of the application to avoid accidentally service other files?  If so, should there be an override for that safety?

**Comment by drforr** [comment]
I'd suggest adding an optional MIME-type variable on the call. Testing a file's magic can be unreliable and slow. Think for example of a 1MB UTF-8 with the only non-ASCII character being a copyright symbol at the end. This way the user can either override bad magic detection or set it for performance purposes.

**Comment by szabgab** [comment]
I'd say by default it should use the same algorithm as the static file serving, (is that based on extension?) but the user can override it with an optional parameter as @drforr suggests.

**Comment by ufobat** [comment]
This makes sense :)

**Comment by drforr** [comment]
Right. I wasn't suggesting avoiding the MIME-type check altogether, I was just suggesting a way to override the behavior. Usually checking magic is fast so it's not a problem, but I've worked with documents where it's unreliable, or sometimes you simply want to, say, upgrade text/ASCII to text/UTF-8 because it's the right thing to do.
